runtime/runtime/src/actions.rs

```diff
@@ -131,16 +118,52 @@ pub(crate) fn execute_function_call(
     if checked_feature!("stable", ChunkNodesCache, protocol_version) {
         runtime_ext.set_trie_cache_mode(TrieCacheMode::CachingChunk);
     }
-    let result = near_vm_runner::run(
-        &code,
+    let result_from_cache = near_vm_runner::run(
+        account,
+        None,
         &function_call.method_name,
         runtime_ext,
-        context,
+        &context,
         &config.wasm_config,
         &config.fees,
         promise_results,
         apply_state.cache.as_deref(),
     );
+    let result = match result_from_cache {
+        Err(VMRunnerError::CacheError(CacheError::ReadError(err)))
+            if err.kind() == std::io::ErrorKind::NotFound =>
+        {
+            let code = match get_contract_code(
+                &runtime_ext,
+                account,
+                apply_state.current_protocol_version,
+            ) {
+                Ok(Some(code)) => code,
+                Ok(None) => {
+                    let error =
+                        FunctionCallError::CompilationError(CompilationError::CodeDoesNotExist {
```

Comment:
This technically happens after the previous place. However, AFAICT if there is a cached `code_hash` compilation result, then it must mean that the code does exist, and thus this error cannot be raised.

So moving this error case down should not affect visible results in any way.